NeoMathEngine/src/CPU/x86/AvxDll.cpp

```diff
@@ -0,0 +1,105 @@
+/* Copyright © 2017-2020 ABBYY Production LLC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+		http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+--------------------------------------------------------------------------------------------------------------*/
+
+#include <common.h>
+#pragma hdrstop
+
+#if FINE_PLATFORM( FINE_DARWIN ) || FINE_PLATFORM( FINE_LINUX )
+#include <cpuid.h>
+#endif
+#if FINE_PLATFORM( FINE_WINDOWS )
+#include <intrin.h>
+#endif
+
+#include <AvxDll.h>
+#include <MathEngineCommon.h>
+#include <MemoryHandleInternal.h>
+
+namespace NeoML {
+
+CAvxDll::CAvxDll() : isLoaded( false ), functionAdresses{}
+{
+	if( !isAvxAvailable() || !Load( libName ) ) {
+		return;
+	}
+
+	loadFunction( TFunctionPointers::BlobConvolution_avx_f3x3_c24_fc24, "BlobConvolution_avx_f3x3_c24_fc24" );
+
+	isLoaded = true;
+}
+
+CAvxDll& CAvxDll::GetInstance()
+{
+	static CAvxDll instance;
+	return instance;
+}
+
+void CAvxDll::loadFunction( TFunctionPointers functionType, const char* functionName )
+{
+	void* functionAdress = reinterpret_cast<void*>( GetProcAddress( functionName ) );
+	ASSERT_EXPR( functionAdress != nullptr );
+
+	functionAdresses[static_cast<size_t>(functionType)] = functionAdress;
+}
+
+void CAvxDll::CallBlobConvolution_avx_f3x3_c24_fc24( IMathEngine& mathEngine, int threadCount, const CCommonConvolutionDesc& desc,
+	const float* sourceData, const float* filterData, const float* freeTermData, float* resultData ) const
+{
+	typedef void ( *FuncType )( IMathEngine& mathEngine, int threadCount, const CCommonConvolutionDesc&, const float*, const float*, const float*, float* );
+	FuncType func = reinterpret_cast<FuncType>( functionAdresses.at( static_cast<size_t>( TFunctionPointers::BlobConvolution_avx_f3x3_c24_fc24 ) ) );
+
+	ASSERT_EXPR( func != nullptr );
```

Comment:
ASSERT_EXPR are used only in MathEngine, you should use assert here.

Reply:
@SAngeliuk This code is part of MathEngine library. It isn't belong to separate avx library.

Reply:
Oh, I see. We've already refactored this issue. Now we should remove all assert(), but of course in the next PR...